R/github-action.R

```diff
@@ -22,15 +22,19 @@ use_github_action <- function(name,
                               ignore = TRUE,
                               open = FALSE) {
 
-  # Append a `.yaml` extension if needed
-  stopifnot(is_string(name))
+  # Check if a custom URL is being used.
+  if (is.null(url)) {
+    stopifnot(is_string(name))
 
-  if (!grepl("[.]yaml$", name)) {
-    name <- paste0(name, ".yaml")
-  }
+    # Append a `.yaml` extension if needed
+    if (!grepl("[.]yaml$", name)) {
+      name <- paste0(name, ".yaml")
+    }
 
-  if (is.null(url)) {
     url <- glue("https://raw.githubusercontent.com/r-lib/actions/master/examples/{name}")
+  } else {
+    stopifnot(is_string(url))
+    stopifnot(grepl(".*[.]yaml$", url))
```

Comment:
I think is too strict, for instance `.yml` is pretty common extension for yaml files, even if `.yaml` is actually the one recommended by the [yaml faq](https://yaml.org/faq.html).

I think we should actually just not check the extension at all.

Reply:
Okay, I've removed the extension check completely for a custom URL. The only enforcement is a string of some type.